### PR TITLE
[JUJU-3524] Update dqlite dep

### PR DESCRIPTION
**Pull request by SimonRichardson**
The following updates the go-dqlite dependency independantly from another PR as there are some required changes because of this update.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc

## QA steps

Test pass in the following package: 

```sh
$ go test -v ./database -check.v
```

Additionally a quick smoke test:

```sh
$ juju bootstrap lxd test --build-agent
$ juju enable-ha
$ juju add-model default
$ juju deploy ubuntu -n 3
```

### CODE REVIEW

**Comment by SimonRichardson** [comment]
@manadart This currently fails:

```
=== RUN   Test
PASS: bootstrap_test.go:27: bootstrapSuite.TestBootstrapSuccess 0.008s
PASS: migration_test.go:19: migrationSuite.TestMigrationSuccess 0.006s
PASS: node_test.go:32: nodeManagerSuite.TestEnsureDataDirSuccess        0.000s

----------------------------------------------------------------------
FAIL: node_test.go:83: nodeManagerSuite.TestIsBootstrappedNode

node_test.go:145:
    c.Check(asBootstrapped, jc.IsTrue)
... obtained bool = false


----------------------------------------------------------------------
PASS: node_test.go:57: nodeManagerSuite.TestIsExistingNode      0.000s

----------------------------------------------------------------------
FAIL: node_test.go:148: nodeManagerSuite.TestSetClusterServersSuccess

node_test.go:187:
    // cluster.yaml should reflect the new server list.
    c.Check(string(data), gc.Equals, `
    - Address: 10.6.6.6:17666
      ID: 3297041220608546238
      Role: 0
    `[1:])
... obtained string = "" +
...     "- Address: 127.0.0.1:17666\n" +
...     "  ID: 3297041220608546238\n" +
...     "  Role: 0\n"
... expected string = "" +
...     "- Address: 10.6.6.6:17666\n" +
...     "  ID: 3297041220608546238\n" +
...     "  Role: 0\n"
... String difference:
...     [0]: "- Address: 127.0.0.1:17666" != "- Address: 10.6.6.6:17666"



----------------------------------------------------------------------
PASS: node_test.go:194: nodeManagerSuite.TestSetNodeInfoSuccess 0.000s
PASS: node_test.go:236: nodeManagerSuite.TestWithAddressOptionSuccess   0.000s
PASS: node_test.go:261: nodeManagerSuite.TestWithClusterOptionSuccess   0.000s
PASS: node_test.go:247: nodeManagerSuite.TestWithTLSOptionSuccess       0.001s
OOPS: 8 passed, 2 FAILED
--- FAIL: Test (0.02s)
FAIL
FAIL    github.com/juju/juju/database   0.047s
FAIL
```

File: database/node_test.go [hunk 1]
```diff
@@ -226,11 +234,10 @@ Role: 0
...
 	// info.yaml should reflect the new node info.
-	c.Check(string(data), gc.Equals, `
-Address: 10.6.6.6:17666
-ID: 3297041220608546238
-Role: 0
-`[1:])
+	var result dqlite.NodeInfo
+	err = yaml.Unmarshal(data, &result)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(result, jc.DeepEquals, server)
```
Comment (SimonRichardson) [inline database/node_test.go:240]: Had to change this so that we don't care about the ordering of the fields.

**Review by manadart — APPROVED** [review]
I assume the failures mentioned are rectified. Passing consistently for me.
Also tested some HA scenarios - all looking good.

**Pull request merged**